### PR TITLE
feat: update translation with default i18n

### PR DESCRIPTION
**Pull request by caalador**
Update documentation to define
the default usage of i18n and
end with the custom translation
provider functionality.

part of vaadin/flow#5917

### CODE REVIEW

File: articles/advanced/i18n-localization.asciidoc [hunk 1]
```diff
@@ -7,9 +7,59 @@ order: 20
...
+To use localization and translation strings, the application only needs to have the translation properties available on the classpath under the directory `vaadin-i18n` with the filename prefix `translations` e.g. `src/main/resources/vaadin-i18n/translations.properties`.
 
-== Defining the I18n Provider Property
+When <<Using Localization in the Application, calling for translation>>, or the I18NProvider is gotten for the first time, the folder resource `vaadin-i18n` will be checked if it contains any `translations.properties` or `translation_[langcode].properties` files.
+Any lang codes will be collected from the available properties files and added automatically as provided locales in the [classname]`DefaultI18NProvider`.
+
+The file `translations.properties` is a default translation file that will be used for any [classname]`Locale` that doesn't have a specific translations file.
+Locale translation files are made as [filename]`translations_fi_FI.properties` or [filename]`translations_en_GB.properties`
```
Comment (mcollovati) [inline articles/advanced/i18n-localization.asciidoc:16]: Is `are made` intended, or should it be `are named`?

Comment (mcollovati) [reply articles/advanced/i18n-localization.asciidoc:16]: We can maybe add also an example without country code, e.g. `translations_de.properties`

Comment (caalador) [reply articles/advanced/i18n-localization.asciidoc]: Fixed and added Language sample also added note on supported 1-3 parts.

File: articles/advanced/i18n-localization.asciidoc [hunk 2]
```diff
@@ -7,9 +7,60 @@ order: 20
 
 = Localization
 
-To use localization and translation strings, the application only needs to implement [interfacename]`I18NProvider` and define the fully qualified class name in the property `i18n.provider`.
+To use localization and translation strings, the application only needs to have the translation properties available on the classpath under the directory `vaadin-i18n` with the filename prefix `translations` e.g. `src/main/resources/vaadin-i18n/translations.properties`.
 
-== Defining the I18n Provider Property
+When <<Using Localization in the Application, calling for translation>>, or the I18NProvider is gotten for the first time, the folder resource `vaadin-i18n` will be checked if it contains any `translations.properties` or `translation_[langcode].properties` files.
```
Comment (mcollovati) [inline articles/advanced/i18n-localization.asciidoc:12]:
```suggestion
When <<Using Localization in the Application, calling for translation>>, or the I18NProvider is gotten for the first time, the folder resource `vaadin-i18n` will be checked if it contains any `translations.properties` or `translations_[langcode].properties` files.
```

**Review by mcollovati — CHANGES_REQUESTED** [review]

**Review by mcollovati — APPROVED** [review]

**Pull request merged**